Services/Notifications/classes/class.ilNotificationGUI.php

```diff
@@ -139,20 +132,42 @@ public function getOSDNotificationsObject()
 
         $result = new stdClass();
         $result->notifications = $notifications;
-        $result->server_time = time();
-        echo json_encode($result);
-        exit;
+        $toasts = [];
+        foreach ($result->notifications as $notification) {
+            $toast = $this->dic->ui()->factory()->toast()->standard(
+                $notification['data']->title,
+                $this->dic->ui()->factory()->symbol()->icon()->custom($notification['data']->iconPath, ''))
+            ->withAction('ilias.php?' . http_build_query([
+                    'baseClass' => 'ilNotificationGUI',
+                    'cmd' => 'removeOSDNotifications',
+                    'cmdMode' => 'asynch',
+                    'notification_id' => $notification['notification_osd_id']
+            ]))
+            ->withDescription($notification['data']->shortDescription)
+            ->withVanishTime($settings->get('osd_vanish') * 1000)
+            ->withDelayTime($settings->get('osd_delay'));
+            foreach ($notification['data']->links as $link) {
+                $toast = $toast->withAdditionalLink($this->dic->ui()->factory()->link()->standard(
+                    $link->getTitle(),
+                    $link->getUrl()
+                ));
+            }
+            $toasts[] = $toast;
+        }
+        $this->dic->http()->saveResponse(
+            $this->dic->http()->response()
+                ->withBody(Streams::ofString(
+                    $this->dic->ui()->renderer()->render($toasts)
```

Comment:
@iszmais Please keep in mind: Wherever we render the `Toas` elements into the `DOM`, there MUST be a containing element which applies an `aria-live` region: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions

Reply:
Does this really is a requirement for `Toasts` and not for `Notifications`?

Reply:
Maybe for both, but we should handle this for `Toasts` now (means: in this project).